src/public/js/main.js

```diff
@@ -15,16 +17,33 @@
       }
 
       var msg = $('#post').val();
-      var encoded = RS.encode(msg);
+      var i = 1;
+      var last = 0;
+      var encoded = new Array(0);
+      for (; i <= msg.length; i++) {
+        if (i % k == 0) {
+          Array.prototype.push.apply(encoded, RS.encode(msg.slice(i - k, i)));
```

Comment:
Yeah, what I thought the problem was is that push expects comma separated values and not an array as its input. My understanding of what was happening here is that we push each value from RS.encode(msg.slice(i - k, i) on to encoded and that is what the apply does, but admittedly I have no idea what apply does, your message here has actually made it click with me a bit though so thanks! And I think you are right that apply is not needed! I don't know why the source I was reading said this is how you do it :/ I am gonna test without the apply.